### DISCUSSION TITLE
How can I use cusvaer simulator

### DISCUSSION THREAD

**Discussion by tsaizl** [NVIDIA/cuQuantum#50] [Q&A] [answered]
![圖片](https://user-images.githubusercontent.com/132208163/235356258-f8b88a25-54a7-45fc-bcc0-93863ade9abe.png)
This is an example from Nvidia cuquantum
and I want to execute on my device but I can't import cusvaer

**Comment by tsaizl** [comment]
![圖片](https://user-images.githubusercontent.com/132208163/235356744-10d100a3-3017-4a27-bd78-480e62c2f6e5.png)

It will call an error named **no module named 'cusvaer'**

**Comment by leofang** [comment, marked as answer]
Sorry, someone this fell out of the team's radar. You'd need to use the cuQuantum Appliance container for the time being: https://docs.nvidia.com/cuda/cuquantum/appliance/index.html

**Reply by Jookare** [reply]
The page that you linked is 404, also I am using cuQuantum-appliance container that is pulled from the nvidia NGC catalog. Still same error
```
ModuleNotFoundError: No module named 'cusvaer.backends'; 'cusvaer' is not a package
```
when I try to run the example code from [cuQuantum docs](https://docs.nvidia.com/cuda/cuquantum/latest/appliance/cusvaer.html#using-cusvaer).

**Reply by leofang** [reply]
The page has been moved recently, the new url is https://docs.nvidia.com/cuda/cuquantum/latest/appliance/index.html. @Jookare what's the version of the cuQuantum Appliance container that you used?

**Reply by Jookare** [reply]
I see. I am using the cuquantum-appliance-23.03 and this is my workflow to recreate this issue:

```
$ apptainer pull docker://nvcr.io/nvidia/cuquantum-appliance:23.03
```
Then create a file `ghz.py` as in the documentation, run the container and execute the file
```
$ apptainer shell --nv -B $PWD cuquantum-appliance_23.03.sif
$ python ghz.py

Traceback (most recent call last):
  File "ghz.py", line 2, in <module>
    from cusvaer.backends import StatevectorSimulator
  File "/scratch/jookare/ghz.py", line 2, in <module>
    from cusvaer.backends import StatevectorSimulator
ModuleNotFoundError: No module named 'cusvaer.backends'; 'cusvaer' is not a package
```

**Reply by mtjrider** [reply]
Can you run `which python` in the shell?

**Reply by mtjrider** [reply]
~@leofang can you remove the answer tag? @Jookare is using the Appliance container via `apptainer` which produces a singularity image at pull.~ Nevermind. I realized I could do it.

I'd like to confirm that the correct Python environment is being picked up by `apptainer shell ...` ... I suspect it's not calling the `entrypoint.sh` correctly.

**Reply by leofang** [reply]
@mtjrider Note that @tsaizl's question was indeed answered, there's nothing wrong. Ideally, @Jookare should have opened a new discussion thread, not reusing this one. The root cause is different (originally it was simply due to not using container; now it's related to Singularity/Apptainer).

**Reply by mtjrider** [reply]
Ah, @leofang, I missed the original question/answer and thought this thread was a continuation. My mistake.
I created another issue containing @Jookare's comments.
Here: https://github.com/NVIDIA/cuQuantum/issues/76

**Comment by ketan-more-github** [comment]
i want t install cusvaer can anyone help me to install the same

**Reply by ymagchi** [reply]
Hi @ketan-more-github,
Currently cusvaer has been provided inside cuQuantum Appliance container images. Please refer to https://docs.nvidia.com/cuda/cuquantum/latest/getting-started/index.html#running-the-cuquantum-appliance to use the container.
